neqo-client/src/main.rs

```diff
@@ -6,6 +6,7 @@
 
 #![cfg_attr(feature = "deny-warnings", deny(warnings))]
 #![warn(clippy::use_self)]
+#![deny(clippy::pedantic)]
```

Comment:
In other projects, we use `warn(clippy::pedantic)` in code, but then execute pre-commit hooks and CI with `-D warnings`.  I think that is a better approach overall than having this in code.